OTAnalytics/application/use_cases/editor/section_editor.py

```diff
@@ -0,0 +1,138 @@
+import contextlib
+from typing import Callable
+
+from OTAnalytics.application.application import CancelAddSection
+from OTAnalytics.application.use_cases.section_repository import AddSection
+from OTAnalytics.domain import geometry
+from OTAnalytics.domain.geometry import coordinate_from_tuple
+from OTAnalytics.domain.section import (
+    ID,
+    NAME,
+    RELATIVE_OFFSET_COORDINATES,
+    Area,
+    LineSection,
+    MissingSection,
+    Section,
+    SectionId,
+    SectionRepository,
+)
+from OTAnalytics.domain.types import EventType
+
+MetadataProvider = Callable[[], dict]
+
+
+class MissingCoordinate(Exception):
+    pass
+
+
+def validate_coordinates(coordinates: list[tuple[int, int]]) -> None:
+    if not coordinates:
+        raise MissingCoordinate("First coordinate is missing")
+    elif len(coordinates) == 1:
+        raise MissingCoordinate("Second coordinate is missing")
+
+
+def validate_section_information(
+    meta_data: dict, coordinates: list[tuple[int, int]]
+) -> None:
+    validate_coordinates(coordinates)
+    if not meta_data:
```

Comment:
`meta_data`s type signature should be denoted as `dict | None` if it can be `None`.

Reply:
My bad @briemla ! I did not realize before making that comment that the if statement is supposed to check if the dictionary is empty. So, it was correct all along. Thus, I reverted your last comment applying this change request of mine.